packages/widget/src/types/widget.ts

```diff
@@ -228,7 +234,46 @@ export interface WidgetConfig {
     Partial<Record<'internal', string[]>>;
 }
 
-export interface WidgetConfigProps {
+export interface FormFieldOptions {
+  setUrlSearchParam: boolean;
+}
+
+export type FieldNames =
+  | 'fromChain'
+  | 'toChain'
+  | 'fromToken'
+  | 'toToken'
+  | 'fromAmount'
+  | 'toAddress';
+
+export type FieldValues = {
+  fromChain: number | undefined;
+  fromToken: string | undefined;
+  toChain: number | undefined;
+  toToken: string | undefined;
+  fromAmount: number | string | undefined;
+  toAddress: ToAddress | string | undefined;
+};
+
+export type SetFieldValueFunc = <K extends FieldNames>(
+  key: K,
+  value: FieldValues[K],
+  options?: FormFieldOptions,
+) => void;
+
+export type FormFunctions =
+  | {
+      setFieldValue: SetFieldValueFunc;
+    }
+  | undefined;
```

Comment:
Why this can be undefined?

Reply:
Because the initial state of the useRef will be undefined before the functions are added to it

Reply:
I'll change this to

```
export type FormFunctions = {
  setFieldValue: SetFieldValueFunction;
};
```

And point of use should change to

```
const formRef = useRef<FormFunctions | undefined>()
```

That seems cleaner